extension/encoding/azureencodingextension/internal/unmarshaler/logs/category.go

```diff
@@ -0,0 +1,235 @@
+func (r *azureLogRecordBase) GetLevel() (plog.SeverityNumber, string, bool) {
+	if r.Level == nil {
+		return plog.SeverityNumberUnspecified, "", false
+	}
+
+	severity := asSeverity(*r.Level)
+	// Saving original log.Level text,
+	// not the internal OpenTelemetry SeverityNumber -> SeverityText mapping
+	return severity, *r.Level, true
+}
+
+// PutCommonAttributes puts already parsed common attributes into provided Attributes Map/Body
+func (r *azureLogRecordBase) PutCommonAttributes(attrs pcommon.Map, _ pcommon.Value) {
+	// Common fields for all Azure Resource Log Categories should be
+	// placed as attributes, no matter if we can map the category or not
+	unmarshaler.AttrPutStrIf(attrs, unmarshaler.AttributeAzureOperationName, r.OperationName)
+	unmarshaler.AttrPutStrPtrIf(attrs, attributeAzureOperationVersion, r.OperationVersion)
+	unmarshaler.AttrPutStrPtrIf(attrs, attributeAzureResultType, r.ResultType)
+	unmarshaler.AttrPutStrPtrIf(attrs, attributeAzureResultSignature, r.ResultSignature)
+	unmarshaler.AttrPutStrPtrIf(attrs, attributesAzureResultDescription, r.ResultDescription)
+	unmarshaler.AttrPutStrPtrIf(attrs, string(conventions.NetworkPeerAddressKey), r.CallerIPAddress)
+	unmarshaler.AttrPutStrPtrIf(attrs, attributeAzureCorrelationID, r.CorrelationID)
+	unmarshaler.AttrPutIntNumberPtrIf(attrs, attributeAzureDuration, r.DurationMs)
+	if r.Identity != nil {
+		unmarshaler.AttrPutMapIf(attrs, attributeAzureIdentity, *r.Identity)
+	}
+}
+
+// PutProperties puts already attributes from "properties" field into provided Attributes Map/Body
+// MUST be implemented by each specific logCategory structure if "properties" field is expected there
+func (*azureLogRecordBase) PutProperties(_ pcommon.Map, _ pcommon.Value) error {
+	// By default - no "properties", so nothing to do here
+	return nil
+}
+
+// azureLogRecordBase represents a single Azure log following the common schema,
+// but has unknown for us Category
+// In this case we couldn't correctly map properties to attributes and simply copy them
+// as-is to the attributes
+type azureLogRecordGeneric struct {
+	azureLogRecordBase
+
+	Properties json.RawMessage `json:"properties"`
+}
+
+func (r *azureLogRecordGeneric) PutProperties(attrs pcommon.Map, body pcommon.Value) error {
+	var properties map[string]any
+	var errs []error
+
+	if len(r.Properties) == 0 {
+		// Nothing to parse
+		return nil
+	}
+
+	// Trying to parse "properties" as correct JSON object...
+	if err := gojson.Unmarshal(r.Properties, &properties); err != nil {
+		errs = append(errs, err)
+		// If failed - trying to parse using a primitive value
+		var val any
+		if err = gojson.Unmarshal(r.Properties, &val); err == nil {
+			// Parsed, put primitive value as "properties" attribute
+			if err = attrs.PutEmpty(attributesAzureProperties).FromRaw(val); err != nil {
+				errs = append(errs, err)
+				// All attempts above - failed, put unparsable properties to log.Body
+				body.SetStr(string(r.Properties))
+			}
```

Comment:
Should there be a `return nil` here, in the case where properties is a primitive value and was successfully added to attrs? Otherwise we'll return an error below, and log a warning in the caller. If that's intended then can you please add a comment?